### PR TITLE
rebase PR #422

### PR DESCRIPTION
**Pull request by JohnRDOrazio**
Replaces PR #422 linked to https://core.trac.wordpress.org/ticket/50493 .

### CODE REVIEW

File: src/wp-includes/rest-api.php [hunk 1]
```diff
@@ -38,26 +38,55 @@ function register_rest_route( $route_namespace, $route, $args = array(), $overri
...
 	if ( ! did_action( 'rest_api_init' ) ) {
 		_doing_it_wrong(
-			'register_rest_route',
+			__FUNCTION__,
 			sprintf(
-				/* translators: %s: rest_api_init */
-				__( 'REST API routes must be registered on the %s action.' ),
-				'<code>rest_api_init</code>'
+				/* translators: 1: rest_api_init, 2: string value of the namespace, 3: string value of the route. */
+				__( 'REST API routes must be registered on the %1$s action. Instead route \'%3$s\' with namespace \'%2$s\' was not registered on this action.' ),
```
Comment (TimothyBJacobs) [inline src/wp-includes/rest-api.php]: This still needs to have the order of placeholders match their appearance order.

Comment (JohnRDOrazio) [reply src/wp-includes/rest-api.php]: done. Order of strings in a single translation has been set to be preferable to the order of the parameters in all other surrounding translations.

**Comment by JohnRDOrazio** [comment]
@TimothyBJacobs fixed

**Comment by TimothyBJacobs** [comment]
Merged in [r58933](https://core.trac.wordpress.org/changeset/58933).

Thanks for the PR!

**Pull request closed without merging**